File: ts/project/.sdk/tm/py/core/response.py
# ProjectName SDK response

from __future__ import annotations
from utility.voxgig_struct import voxgig_struct as vs


class ProjectNameResponse:
    def __init__(self, resmap=None):
        if resmap is None:
            resmap = {}

        self.status = -1
        s = vs.getprop(resmap, "status")
        if s is not None and isinstance(s, (int, float)):
            self.status = int(s)

        self.status_text = ""
        st = vs.getprop(resmap, "statusText")
        if isinstance(st, str):
            self.status_text = st

        self.headers = vs.getprop(resmap, "headers")

        self.json_func = None
        jf = vs.getprop(resmap, "json")
        if callable(jf):
            self.json_func = jf

        self.body = vs.getprop(resmap, "body")

        self.err = None
        e = vs.getprop(resmap, "err")
        if e is not None:
            self.err = e

File: ts/project/.sdk/tm/lua/core/response.lua
-- ProjectName SDK response

local vs = require("utility.struct.struct")

local Response = {}
Response.__index = Response


function Response.new(resmap)
  local self = setmetatable({}, Response)
  resmap = resmap or {}

  self.status = -1
  local s = vs.getprop(resmap, "status")
  if s ~= nil and type(s) == "number" then
    self.status = math.floor(s)
  end

  self.status_text = ""
  local st = vs.getprop(resmap, "statusText")
  if type(st) == "string" then
    self.status_text = st
  end

  self.headers = vs.getprop(resmap, "headers")

  self.json_func = nil
  local jf = vs.getprop(resmap, "json")
  if type(jf) == "function" then
    self.json_func = jf
  end

  self.body = vs.getprop(resmap, "body")

  self.err = nil
  local e = vs.getprop(resmap, "err")
  if e ~= nil then
    self.err = e
  end

  return self
end


return Response
